rasa_utils/bot_server_channel.py

```diff
@@ -98,95 +98,102 @@ def __init__(
         self.cors_origins = [u'*']
         self.agent = agent
         self.port = port
+        self.preprocessor = preprocessor
 
-    @app.route("/conversations/<cid>/log", methods=["GET"])
-    @check_cors
-    def show_log(self, request, cid):
-        request.setHeader("Content-Type", "application/json")
-        return json.dumps(self.message_store[cid])
-
-    @app.route("/conversations/<cid>/tracker", methods=["GET"])
-    @check_cors
-    def tracker(self, request, cid):
-        tracker = self.agent.tracker_store.get_or_create_tracker(cid)
-        tracker_state = tracker.current_state(
-            should_include_events=True,
-            only_events_after_latest_restart=True
-        )
+    @classmethod
+    def name(cls):
+        return "chatroom"
+
+    def blueprint(self, on_new_message):
+        bot_server_webhook = Blueprint('bot_server_webhook', __name__)
+        CORS(bot_server_webhook)
 
-        request.setHeader("Content-Type", "application/json")
-        return json.dumps(tracker_state)
+        @bot_server_webhook.route("/", methods=['GET'])
+        def static():
+            if self.static_files is None:
+                return NoResource()
+            else:
+                return File(self.static_files)
+
+        @bot_server_webhook.route("/health", methods=["GET"])
+        def health():
+            return "healthy"
+
+        @bot_server_webhook.route("/webhook", methods=['POST'])
+        def receive():
+            sender_id = self._extract_sender(request)
+            text = self._extract_message(request)
+            should_use_stream = utils.bool_arg("stream", default=False)
+
+            if should_use_stream:
+                return Response(
+                        self.stream_response(on_new_message, text, sender_id),
+                        content_type='text/event-stream')
+            else:
+                collector = CollectingOutputChannel()
+                on_new_message(UserMessage(text, collector, sender_id))
+                return json.dumps(collector.messages)
 
-    @app.route("/conversations/<cid>/say", methods=["GET"])
-    @check_cors
-    def say(self, request, cid):
-        message, = request.args.get(b"message", [])
-        _payload = request.args.get(b"payload", [])
-        _display_name = request.args.get(b"display_name", [])
-        _uuid = request.args.get(b"uuid", [])
-        logger.info(message)
+        @bot_server_webhook.route("/conversations/<cid>/log", methods=["GET"])
+        def show_log(cid):
+            return json.dumps(self.message_store[cid])
 
-        if len(_display_name) > 0:
-            display_name, = _display_name
+        @bot_server_webhook.route("/conversations/<cid>/tracker", methods=["GET"])
+        def tracker(cid):
             tracker = self.agent.tracker_store.get_or_create_tracker(cid)
```

Comment:
I assume we should check for the availability of an agent as a precaution. `self.agent` will not be available if the channel was started using the new `credentials.yml` method.